prov/usnic/src/usdf_cq.c

```diff
@@ -149,12 +149,106 @@ usdf_cq_readerr_soft(struct fid_cq *fcq, struct fi_cq_err_entry *entry,
 	return 1;
 }
 
+static inline ssize_t
+usdf_cq_sread_common(struct fid_cq *fcq, void *buf, size_t count, const void *cond,
```

Comment:
nothing in this function looks at `count`, so this logic could easily overrun the user's provided buffer